pkg/mongoproxy/plugins/schema/schema.go

```diff
@@ -155,7 +157,10 @@ func (p *SchemaPlugin) Process(ctx context.Context, r *plugins.Request, next plu
 		for _, document := range cmd.Documents {
 			if err := schema.ValidateInsert(ctx, cmd.Database, cmd.Collection, document); err != nil {
 				schemaDeny.WithLabelValues(cmd.Database, cmd.Collection, r.CommandName).Inc()
-				return mongoerror.DocumentValidationFailure.ErrMessage(err.Error()), nil
+				if !p.conf.EnforceSchemaLogOnly {
+					return mongoerror.DocumentValidationFailure.ErrMessage(err.Error()), nil
+				}
+				logrus.Warningf("ENFORCE SCHEMA LOGONLY: %s, in collection: %s", err.Error(), cmd.Collection)
```

Comment:
just notice we have metrics `schemaDeny.WithLabelValues` at line 159 where it provides DB, Collection and cmd info. If your log provides the similar info (Collection is added in log right now), we can use metrics to reduce duplicated info

Reply:
so probably we can have the config name as `enforceSchema`, we return err when it's true; we can add another panel in mongogate metrics `https://grafana.s.wish.site/d/UirYASMnk/mongogate-mongoproxy-native?orgId=1&refresh=1m` <- we need to have this metrics anyway 😉

Reply:
Agree that we need add the new metrics with panel, but it is not conflict with adding them through log. With info in log is better&faster to track and fix. Also this "EnforceSchemaLogOnly" is just for rolling out, and we gonna roll out collection by collection, which means if do have some corner cases that trigger the log, we will fix it and enable other collection after. It won't create lots of overhead on log right, correct me if I'm wrong.

Reply:
either way will be fine. if you prefer both metrics and logs, could you add `db` and `command` info in your logs?

Reply:
added